internal/kube/securedaccess/access.go

```diff
@@ -639,3 +644,27 @@ func hasSecuredAccessLabel(obj *metav1.ObjectMeta) bool {
 	_, ok := obj.Labels["internal.skupper.io/secured-access"]
 	return ok
 }
+
+func ensureOwnerReferences(meta *metav1.ObjectMeta, owners []metav1.OwnerReference) bool {
+	byUID := make(map[string]int, len(owners))
+	for iRef, ref := range owners {
+		byUID[string(ref.UID)] = iRef
+	}
+
+	changed := false
+	for i, ref := range meta.OwnerReferences {
+		uid := string(ref.UID)
+		if _, ok := byUID[uid]; ok {
+			delete(byUID, uid)
+			continue
+		}
+		meta.OwnerReferences = slices.Delete(meta.OwnerReferences, i, i+1)
```

Comment:
Probably not a real or valid scenario, but if for any reason `meta.OwnerReferences` has 2 entries to be removed, this is going to crash.